src/lib/protocols/bittorrent.c

```diff
@@ -105,11 +105,14 @@ static void ndpi_add_connection_as_bittorrent(struct ndpi_detection_module_struc
 					      int bt_offset, int check_hash,
 					      ndpi_confidence_t confidence) {
   struct ndpi_packet_struct *packet = &ndpi_struct->packet;
+  u_int16_t master;
+
+  master = flow->detected_protocol_stack[1] ? flow->detected_protocol_stack[1] : flow->detected_protocol_stack[0];
```

Comment:
It is probably more error prone if this line would be a API function.

Reply:
Ok, I'll create a new wrapper